harness/determined/keras/_tf_keras_trial.py

```diff
@@ -507,7 +506,7 @@ def _configure_callbacks(self, user_callbacks: Optional[List]) -> None:
         )
         callbacks = [self.multiplexer]
 
-        if self.hvd_config.use:
+        if self.context.distributed.size > 1 and self.use_horovod:
```

Comment:
as with estimator trial, I'd prefer you put an assert somewhere that only allows dtrain tfkeras when horovod is in use.

If we ever support non-horovod dtrain tfkeras then we'll already have to go through the code with a fine-tooth comb, and these if statements where we try to predict what the right behavior would be if we supported non-horovod dtrain tfkeras would be more likely to cause quiet bugs than to be helpful.